crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs

```diff
@@ -177,11 +181,13 @@ impl Simulator {
         }
     }
 
-    fn zip_l2_blocks_with_da_blocks(
+    fn calculate_da_blocks(
         &self,
         da_recording_rate: usize,
+        da_finalization_rate: usize,
         fullness_and_bytes: &Vec<(u64, u64)>,
     ) -> Vec<Option<Vec<RecordedBlock>>> {
+        let nones = std::iter::repeat(None).take(da_finalization_rate);
```

Comment:
```suggestion
        let l2_blocks_with_no_da_blocks = std::iter::repeat(None).take(da_finalization_rate);
```

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2240/commits/ff0db6e70e8174fceb5484d13bb9e5ae9a917f68